crates/core/src/globals.rs

```diff
@@ -442,47 +442,63 @@ fn build_memory(context: &JSContextRef) -> anyhow::Result<JSValueRef> {
             let data = data.as_bytes()?;
             let m = extism_pdk::Memory::from_bytes(data)?;
             let mut mem = HashMap::new();
-            let offset = JSValue::Int(m.offset() as i32);
-            let len = JSValue::Int(m.len() as i32);
+            let offset = JSValue::Float(m.offset() as f64);
```

Comment:
Looks good and makes sense to me, but only really in the context of this PR. I think a single comment here about why we're packing the offset into a float may be helpful.

Reply:
Excellent point – added a note!